### DISCUSSION TITLE
Install command

### DISCUSSION THREAD

**Discussion by rjoleary** [u-root/u-root#2120] [Ideas]
The proposed install command creates a bootable drive containing u-root. For example, an invocation might look like:

```
install -kernel=/path/to/kernel -dev=/dev/sdb
```

This would:

- Create a filesystem with the necessary bootable bits on the drive.
- Write the output of the u-root build command to the drive. Extra flags are passed through.
- Copy the specified kernel to the drive.

The important key is the installed drive is sufficient to bootstrap the `install` command. For example, the first invocation of `install` runs on machine A and creates a live USB drive. The USB drive can be used to boot machine B. Finally, while booted from the USB drive, `install` is run a second time to install u-root from the USB drive to the hard drive of machine B.

To start, the devices I had in mind were Minnows and NUCs. Similar things which already exist are:

* Nichrome
* gokrazy

**Comment by pwFoo** [comment]
install command would be great and a additional output `-format=???` to build and write direct to a disk / usb stick?
So u-root could be installed if booted as "live system" by command and installed to a disk / usb stick by `-format=*`
Asked for a similar feature `-format=iso` in #758
